aten/src/ATen/core/MT19937RNGEngine.h

```diff
@@ -126,16 +131,19 @@ class mt19937_engine {
 private:
   uint64_t seed_;
   int left_;
+  bool seeded_;
   uint32_t next_;
-  uint32_t state_[MERSENNE_STATE_N];
+  std::array<uint32_t, MERSENNE_STATE_N> state_;
```

Comment:
I can think of plausible reasons for the changes here, but since there isn't commentary about it in comments anywhere, I'm left guessing. I think the changes are OK, but can you spell out why?

Reply:
This is because of supporting BC set_rng_state. Since THGeneratorState has all the members of the mt19937, I needed to copy in the raw array and other members. Hence, created a constructor and std::array was the way to go to have it as a parameter in the constructor.